File: resize_icons.js
import sharp from 'sharp';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const publicDir = path.join(__dirname, 'public');

async function resizeImage(filename, size) {
    const filePath = path.join(publicDir, filename);
    try {
        const buffer = await sharp(filePath)
            .resize(size, size)
            .toBuffer();
        await sharp(buffer).toFile(filePath);
        console.log(`Resized ${filename} to ${size}x${size}`);
    } catch (error) {
        console.error(`Error resizing ${filename}:`, error);
    }
}

resizeImage('pwa-192x192.png', 192);
resizeImage('pwa-512x512.png', 512);

File: resize_icons.py
from PIL import Image
import os

def resize_image(path, size):
    try:
        img = Image.open(path)
        img = img.resize(size, Image.Resampling.LANCZOS)
        img.save(path)
        print(f"Resized {path} to {size}")
    except Exception as e:
        print(f"Error resizing {path}: {e}")

public_dir = '/home/fk/.gemini/antigravity/scratch/med-tracker/public'
resize_image(os.path.join(public_dir, 'pwa-192x192.png'), (192, 192))
resize_image(os.path.join(public_dir, 'pwa-512x512.png'), (512, 512))
